tests/src/CoreMangLib/cti/system/gc/gckeepalive.cs

```diff
@@ -44,13 +44,13 @@ public bool PosTest1()
             GC.WaitForPendingFinalizers();
             GC.Collect();
```

Comment:
This pattern is [known to be flaky](https://github.com/dotnet/coreclr/issues/4093#issuecomment-296853982). I addressed a few tests that rely on this pattern here: https://github.com/dotnet/coreclr/pull/11216. If this test isn't flaky, I don't think there's any reason to address it here, but as it's written it's possible for the test to fail if we take a really unlucky path through the scheduler.

Reply:
Interesting. No, flakiness isn't the issue I was seeing; it's that the lifetime of `tc` was simply incorrect.